PackagingProjects/RepackageCSharpPluginHelper.ps1

```diff
@@ -0,0 +1,198 @@
+{
+    $sourceDir = GetWorkingDir
+    return [System.IO.Path]::Combine($sourceDir, "TempJarDir")
+}
+
+function GetBuildOutputDir()
+{
+    $sourceDir = GetWorkingDir
+    return [System.IO.Path]::Combine($sourceDir, "../", "DeploymentArtifacts", "CSharpPlugingPayload", $configuration)
+}
+
+
+##### Preparig the existing jar ##########
+
+#
+# Gets the existing jar file. This should be in the working directory.
+#
+function GetSourceJarPath
+{
+    
+    $sourceDir = GetWorkingDir
+    $sourceJarPath = [System.IO.Directory]::GetFiles($sourceDir, "*.jar")
+
+    if ($sourceJarPath.Count -eq 0)
+    {
+        throw "Before using this script please copy a CSharp plugin (the jar file) to the same directory as the script: $sourceDir"
+    }
+
+    if ($sourceJarPath.Count > 1)
+    {
+        throw "Too many jar files in the same directory as the script. Expecting only one"
+    }
+
+    WriteMessage "Found the source plugin jar file: $sourceJarPath"
+
+    return $sourceJarPath
+}
+
+
+
+#
+# Unzips the existing jar file to a well known directory 
+#
+function UnzipJar
+{
+    param([Parameter(Mandatory=$true)][ValidateNotNullOrEmpty()][string]$sourceJarPath)
+    
+    $tempJarDir = GetTempJarFolder
+    if ([System.IO.Directory]::Exists($tempJarDir))
+    {
+        Write-Host -ForegroundColor Gray "Removing old $tempJarDir"
+        Remove-Item $tempJarDir -Recurse
+    }
+
+    try
+    {
+        [System.IO.Compression.ZipFile]::ExtractToDirectory($sourceJarPath, $tempJarDir)
+    }
+    catch [System.IO.InvalidDataException]
+    {
+        throw "A jar file was found but it not apear to be an archive:  $sourceJarPath"
+    }
+
+    WriteMessage "Unzipped the jar file to $tempJarDir"
+}
+
+
+#
+# Check and delete the existing payload from the unzipped jar 
+#
+function DeleteExistingPayload
+{
+    $tempJarDir = GetTempJarFolder
+    $tempPayloadZipPath = [System.IO.Path]::Combine($tempJarDir, "static", $payloadFilename);
+
+    if (![System.IO.File]::Exists($tempPayloadZipPath))
+    {
+        throw "Could not find $payloadFilename inside the C# plugin jar. Did the plugin structure change?"
+    }
+
+    [System.IO.File]::Delete($tempPayloadZipPath)
+
+    WriteMessage "Deleted the existing payload"
+
+    return $tempPayloadZipPath
+   
+}
+
+#
+# Orchestrate finding and unzipping the source jar and deleting the payload from it
+#
+function PrepareExistingJar
+{
+    $sourceJarPath = GetSourceJarPath
+    UnzipJar $sourceJarPath
+    return DeleteExistingPayload 
+}
+
+
+##### Create the new jar #####
+
+#
+# Ensures the specified config build exists by checking the output dir
+#
+function EnsureBuild
+{
+    param([Parameter(Mandatory=$true)][ValidateNotNullOrEmpty()][string]$configuration)
+
+    if (($configuration -ne "Debug") -and ($configuration -ne "Release"))
+    {
+        throw "$configuration is an invalid build configuration. Expecting Debug or Release"
+    }
+
+    # TODO: actually invoke msbuild ? 

```

Comment:
It might be simpler and more convenient to call the script automatically from one of the packaging projects so we get an updated jar on every build (with a warning rather than an error if the source jar isn't found).